go: Containarising a simple golang application

Yuggupta27 opened the pull request:
The driver program utilizes functions from
local packages to provide required output.
This golang app is containarised using a
Dockerfile.

Signed-off-by: Yug Gupta <ygupta@redhat.com>

Yuggupta27 committed:
go: Containarising a simple golang application

The driver program utilizes functions from
local packages to provide required output.
This golang app is containarised using a
Dockerfile.

Signed-off-by: Yug Gupta <ygupta@redhat.com>

Madhu-1 commented on yug-gupta/golang-docker-application/Dockerfile:
```diff
@@ -0,0 +1,4 @@
+FROM golang:1.8.5-jessie
+WORKDIR /go/src
+COPY golang-docker-application /go/src/golang-docker-application
+CMD ["go","run","golang-docker-application/main.go"]
```
you should not copy the source code to docker  image

Madhu-1 commented on yug-gupta/golang-docker-application/Dockerfile:
```diff
@@ -0,0 +1,4 @@
+FROM golang:1.8.5-jessie
```
why 1.8.5?

Madhu-1 commented on yug-gupta/golang-docker-application/main.go:
```diff
@@ -0,0 +1,13 @@
+
+import (
+	"fmt"
+	"ocs-interns/yug-gupta/golang-docker-application/localpackages" // local package with hello() and world() functions
```
is this valid import?

Madhu-1 replied:
missing `github.com/sp98` ?

Yuggupta27 committed:
go: Optimized Docker Image

-> Instead of adding source code, locally
   built binaries are used.
-> scratch base image is used to optimize
   image size

Signed-off-by: Yug Gupta <ygupta@redhat.com>
